EFCore.SqlServer.HierarchyId.Test/Migrations/NeedsBuildReferencesFixture.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.EntityFrameworkCore.TestUtilities;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.EntityFrameworkCore.SqlServer.Migrations
+{
+    /// <summary>
+    /// use this class fixture when you need to get build references
+    /// and are compiling against .net core & dotnetframework
+    /// </summary>
+    public class NeedsBuildReferencesFixture
+    {
+        public NeedsBuildReferencesFixture()
+        {
+            AssemblyUtils.SetEntryAssembly();
+        }
+
+        private IEnumerable<BuildReference> _getRequiredReferences()
+        {
+            if (AssemblyUtils.WasEntryAssemblyUpdated)
```

Comment:
This seems like overkill. Do you really need to update the entry assembly? Can't we just use `#if NET472`?

Reply:
I modified the entry assembly to get the DependencyContext.Default to not be null (referenced in BuildReference.cs), without having to make changes to the code from dotnet/efcore (everything in the TestUtilities folder).  I've tried experimenting without it, and ran into referencing issues, but can dive deeper if necessary.